search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/LlmIOUtil.java

```diff
@@ -57,11 +60,19 @@ public static ChatCompletionInput createChatCompletionInput(
         String question,
         List<Interaction> chatHistory,
         List<String> contexts,
-        int timeoutInSeconds
+        int timeoutInSeconds,
+        String llmResponseField
     ) {
-        Llm.ModelProvider provider = Llm.ModelProvider.OPENAI;
-        if (llmModel != null && llmModel.startsWith(BEDROCK_PROVIDER_PREFIX)) {
-            provider = Llm.ModelProvider.BEDROCK;
+        Llm.ModelProvider provider = null;
+        if (llmResponseField == null) {
+            provider = Llm.ModelProvider.OPENAI;
+            if (llmModel != null) {
+                if (llmModel.startsWith(BEDROCK_PROVIDER_PREFIX)) {
+                    provider = Llm.ModelProvider.BEDROCK;
+                } else if (llmModel.startsWith(COHERE_PROVIDER_PREFIX)) {
+                    provider = Llm.ModelProvider.COHERE;
```

Comment:
add test?

Reply:
I do have tests for the new code.

Reply:
I don't see any test for cohere in the `LlmIOUtilTests`

Reply:
Hmm.. This is covered in the IT tests I added, but I added UT to cover this.